### PR TITLE
Added Exporter and Unit Test for Map::RowVectorX<t>

### PR DESCRIPTION
**Pull request by coatless**
Per SO:

http://stackoverflow.com/questions/36920689/rowvectorxd-not-supported-in-rcppeigen

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Thanks, James -- that is very helpful (and very _fast_).  Paging @yixuan to take a peek too but I guess we should be able to fold this in.

**Comment by yixuan** [comment]
It looks pretty nice, thank you @coatless !

Could you please also convert tab to 4 spaces in `RcppEigenWrap.h` and make the indentation consistent with the context in `runit.RcppEigen.R`? (It is tiny issue but would be nice of you to do so.)

**Comment by eddelbuettel** [comment]
Good point tabs (I tend to avoid them) and four spaces.  The Rcpp repo has some emacs config, it needed.

**Comment by coatless** [comment]
@yixuan done.

File: inst/include/RcppEigenWrap.h [hunk 1]
```diff
@@ -217,6 +217,21 @@ namespace Rcpp{
...
+          Rcpp::Vector<RTYPE> vec ;
+          
+        public:
+          Exporter(SEXP x) : vec(x) {
+            if (TYPEOF(x) != RTYPE)
+              throw std::invalid_argument("Wrong R type for mapped rowvector");
+          }
+          OUT get() {return OUT(vec.begin(), vec.size());}
+        } ;
+
```
Comment (eddelbuettel) [inline inst/include/RcppEigenWrap.h]: That whole block is two-space-intended (after the initial four) but c'est la vie now, we'll get it next time or maybe I'll fix it tomorrow.

**Pull request merged**